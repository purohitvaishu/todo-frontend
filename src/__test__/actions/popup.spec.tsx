import {
  createTodo,
  editTodo,
  deleteTodo,
  getTodo,
  callPopup
} from "../../actions/popup.actions";
import { data, popup } from "../../mockData/action.data";

describe("Popup: action", () => {
  describe("for create", () => {
    test("call api", () => {
      expect(createTodo(data)).toMatchSnapshot();
    });
  });

  describe("for edit", () => {
    test("call api", () => {
      expect(editTodo(data, 1)).toMatchSnapshot();
    });
  });

  describe("for delete", () => {
    test("call api", () => {
      expect(deleteTodo(1)).toMatchSnapshot();
    });
  });

  describe("for get", () => {
    test("call api", () => {
      expect(getTodo()).toMatchSnapshot();
    });
  });

  describe("for popup", () => {
    test("call function", () => {
      expect(callPopup(popup)).toMatchSnapshot();
    });
  });
});
