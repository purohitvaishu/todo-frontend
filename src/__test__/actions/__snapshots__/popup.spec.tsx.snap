// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup: action for create call api 1`] = `
Object {
  "body": Object {
    "bucketname": "The Godfather",
    "completed": false,
    "description": "Submit to doctor",
    "title": "File",
  },
  "endPoint": "api/todos/",
  "method": "post",
  "type": "app/CALL_API",
  "types": Array [
    "actions/popup/ADD_LIST_REQUEST",
    "actions/popup/ADD_LIST_SUCCESS",
    "actions/popup/ADD_LIST_FAILED",
  ],
}
`;

exports[`Popup: action for delete call api 1`] = `
Object {
  "endPoint": "api/todos/1/",
  "method": "delete",
  "type": "app/CALL_API",
  "types": Array [
    "actions/popup/DELETE_LIST_REQUEST",
    "actions/popup/DELETE_LIST_SUCCESS",
    "actions/popup/DELETE_LIST_FAILED",
  ],
}
`;

exports[`Popup: action for edit call api 1`] = `
Object {
  "body": Object {
    "bucketname": "The Godfather",
    "completed": false,
    "description": "Submit to doctor",
    "title": "File",
  },
  "endPoint": "api/todos/1/",
  "method": "put",
  "type": "app/CALL_API",
  "types": Array [
    "actions/popup/EDIT_LIST_REQUEST",
    "actions/popup/EDIT_LIST_SUCCESS",
    "actions/popup/EDIT_LIST_FAILED",
  ],
}
`;

exports[`Popup: action for get call api 1`] = `
Object {
  "endPoint": "api/todos/",
  "method": "get",
  "type": "app/CALL_API",
  "types": Array [
    "actions/popup/GET_LIST_REQUEST",
    "actions/popup/GET_LIST_SUCCESS",
    "actions/popup/GET_LIST_FAILED",
  ],
}
`;

exports[`Popup: action for popup call function 1`] = `
Object {
  "data": Object {
    "edit": undefined,
    "params": Object {
      "bucketname": "Fullfill todo list",
      "completed": true,
      "description": "Bucket for testing",
      "id": 2,
      "title": "Bucket",
    },
    "value": undefined,
  },
  "type": "actions/popup/APP_LIST_POPUP",
}
`;
